File: intermediario2/SBC 2001/allYouNeedIsLove.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){

    int n, a, b, i = 0, c;
    string sa, sb;
    scanf("%d", &n);

    while(i < n){
        cin >> sa;
        cin >> sb;
        a = stoi(sa, nullptr, 2);
        b = stoi(sb, nullptr, 2);
        c = __gcd(a, b);
        if(c > 1) printf("Pair #%d: All you need is love!\n", i+1);
        else printf("Pair #%d: Love is not all you need!\n", i+1);
        i++;
    }


    return 0;
}

File: intermediario2/SBC 2001/allYouNeedIsLove.py
def gcd(x, y): 
  
    if x > y: 
        small = y 
    else: 
        small = x 
    for i in range(1, small+1): 
        if((x % i == 0) and (y % i == 0)): 
            gcd = i 
              
    return gcd 

n = int(raw_input())
i = 0



while (i < n):
    a = int(raw_input(), 2)
    b = int(raw_input(), 2)

    if gcd(a, b) != 1 : print ("Pair #%s: All you need is love!" % str(i+1))
    else: print ("Pair #%s: Love is not all you need!" % str(i+1))
    i+=1